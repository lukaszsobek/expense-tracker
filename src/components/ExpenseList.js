import React from "react"
import { connect } from "react-redux"

import ExpenseListItem from "./ExpenseListItem"
import { getVisibleExpenses } from "../selectors"

export const ExpenseList = ({ expenses }) => {

    if(!expenses) {
        return <div>No expenses defined</div>
    }

    return (
        <div>
            {expenses.map(expense => (
                <ExpenseListItem
                key={expense.id}
                {...expense}
                />
            ))}
        </div>
    )
}

const mapStateToProps = state => ({
    expenses: getVisibleExpenses(state.expenses, state.filters)
})

export default connect(mapStateToProps)(ExpenseList)
