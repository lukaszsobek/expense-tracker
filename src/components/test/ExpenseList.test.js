import React from "react"
import { shallow } from "enzyme"

import { ExpenseList } from "../ExpenseList"

describe("Renders ExpenseList",() => {

    it("renders without expenses",() => {
        const wrapper = shallow(<ExpenseList />)
        expect(wrapper).toMatchSnapshot()
    })

    it("renders with expenses",() => {
        const expenses = [
            {id:1, test: "test"},
            {id:2, blue: "green"},
            {id:3, isLast: true}
        ]

        const wrapper = shallow(<ExpenseList expenses={expenses} />)
        expect(wrapper).toMatchSnapshot()
    })
})