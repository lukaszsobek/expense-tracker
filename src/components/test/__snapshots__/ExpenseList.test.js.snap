// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ExpenseList renders with expenses 1`] = `
<div>
  <ExpenseListItem
    id={1}
    key="1"
    test="test"
  />
  <ExpenseListItem
    blue="green"
    id={2}
    key="2"
  />
  <ExpenseListItem
    id={3}
    isLast={true}
    key="3"
  />
</div>
`;

exports[`Renders ExpenseList renders without expenses 1`] = `
<div>
  No expenses defined
</div>
`;
